elastic/kibana
# [Search] Allow users to save mappings with errors

## Summary

This adds a callout to the index overview page if there's an error with the semantic text mappings. It also allows users to force saving when facing errors deploying a model associated with an inference endpoint, and has minor bugfix for referencing a text field with child fields in a semantic text field.
<img width="1058" alt="Screenshot 2024-07-15 at 18 19 15" src="https://github.com/user-attachments/assets/903878ef-45bf-4a50-9760-9ae62842853b">

<img width="659" alt="Screenshot 2024-07-15 at 18 29 13" src="https://github.com/user-attachments/assets/5ca50fa1-dbdd-406e-9b48-7ffeb4cd0315">
<img width="752" alt="Screenshot 2024-07-15 at 18 29 07" src="https://github.com/user-attachments/assets/dfd18199-f0d8-4f13-ac44-e9b91a8ef492">

Testing:

- Create an index
- Add a text field and semantic text field referencing an existing inference endpoint (eg: elser_model_2) using the UI
- Save mappings
- Now delete the endpoint using the dev console or UI (`DELETE /_inference/sparse_embeddings/elser_model_2`) for elser_model_2.
- Go to that index's overview page (or refresh)
- You should see a callout telling you that there are errors in this index
